sunpy/map/sources/sdo.py

```diff
@@ -115,4 +117,32 @@ def observatory(self):
     @classmethod
     def is_datasource_for(cls, data, header, **kwargs):
         """Determines if header corresponds to an HMI image"""
-        return str(header.get('instrume', '')).startswith('HMI')
+        return (str(header.get('TELESCOP', '')).endswith('HMI') and
+                not HMISynopticMap.is_datasource_for(data, header))
+
+
+class HMISynopticMap(HMIMap):
+
+    def __init__(self, data, header, **kwargs):
+        super().__init__(data, header, **kwargs)
+
+        if self.meta['cunit1'] == 'Degree':
+            self.meta['cunit1'] = 'degree'
+
+        if self.meta['cunit2'] == 'Sine Latitude':
+            self.meta['cunit2'] = 'degree'
+
+            # Since, this map uses the cylindrical equal-area (CEA) projection,
+            # the spacing should be modified to 180/pi times the original value
+            # Reference: Section 5.5, Thompson 2006
+            self.meta['cdelt2'] = 180 / np.pi * self.meta['cdelt2']
+            self.meta['cdelt1'] *= -1
```

Comment:
Does anyone know why this change needs to be made? A negative CDELT seems valid to me...

Reply:
Ugh.  I think it's quite likely that the CDELT1 is intentionally negative, so it'd be incorrect to invert it like this.  The Carrington longitude of disk center decreases over time (i.e., from 360 degrees to 0 degrees), so it'd be reasonable to construct a synoptic map with a decreasing horizontal axis.

The gallery example boldly claims that `CDELT` has the "wrong sign" without any justification.

Reply:
On the other hand, it appears that the horizontal axis for the image provided by HMI (http://jsoc.stanford.edu/data/hmi/synoptic/hmi.Synoptic_Mr.2191.png):
![hmi Synoptic_Mr 2191](https://user-images.githubusercontent.com/991759/80215327-4003b080-860a-11ea-88ce-956d516629f7.png)
matches our gallery example (which negates CDELT1):
![sphx_glr_hmi_synoptic_maps_001](https://user-images.githubusercontent.com/991759/80215446-73463f80-860a-11ea-9f88-5f3f91710d10.png)

So maybe HMI *does* have the incorrect sign for CDELT1 in their header?  Do we have written confirmation of this?

Reply:
<sigh> I don't know why their CDELT1 needs to be inverted, but it definitely needs to be inverted to get the right answers (see the magnetograms below).  Let's document this assumption explicitly in our docstring.

Here's the magnetogram when the Carrington longitude of disk center is ~90 degrees:
![shmi_maglc_fd_20170615_224645](https://user-images.githubusercontent.com/991759/80216232-be148700-860b-11ea-8391-7617251e6834.png)

Here's the magnetogram when the Carrington longitude of disk center is ~300 degrees:
![shmi_maglc_fd_20170531_143444](https://user-images.githubusercontent.com/991759/80216400-ffa53200-860b-11ea-83d4-859bd7b2fadb.png)

Reply:
That is, their provided images show the correct axis values, which is not consistent with their CDELT1 sign.

Reply:
> That is, their provided synoptic images show the correct axis values, which is not consistent with their CDELT1 sign.

iirc this is why it's `*-1`. Because otherwise it didn't match their published synoptic images.

Reply:
Thanks for checking this out @ayshih. I am quite reluctant to do `self.meta['cdelt1'] *= -1` without cross checking this against something else in the header that indicates we should do this. It could be possible that JSOC fix the CDELT value in the future and then we end up erroneously multiplying by -1. To be on the safe side I would say we shouldn't make this fix for now (it's easy enough to add in the future), but wonder what others think?

Reply:
Er, not fixing the CDELT and providing maps with coordinates that don't match publicly released plots sounds like a bad idea.

We can protect against them fixing the sign in their headers if it were instead `self.meta['cdelt1'] = np.abs(self.meta['cdelt'])`.

Reply:
👍 that sounds like a good solution

Reply:
If you are interested, [here is the code](http://jsoc.stanford.edu/data/hmi/synoptic/displaysynoptic.pro) that generates the HMI synoptic maps (apologies for the code quality, I wrote it back in my IDL days).